### PR TITLE
OCP 4

### PR DESCRIPTION
**Pull request by ben-alkov**
This PR intends to update all of OSBS-Box to work with OpenShift 4.

tl;dr, “New” OSBS-Box is not intended to run locally, and assumes a multi-tenant
cluster, to which you have cluster admin credentials, and which either
*has* Filesystem-capable PVs, or can be so configured.

N.B. that the resulting deployment of OSBS-Box will *not* work. Everything
currently available (excepting Cachito) deploys and runs successfully. Builds
*can* be triggered, but **at least** two things are preventing successful
builds: CustomBuildStrategy and Basic Authentication. OSBS itself is currently
being re-architected to address these issues; stay tuned for more.

“Old” OSBS-Box assumed a development OpenShift 3 cluster running locally in a VM
(e.g. instantiated via `oc cluster up`). The only assumptions made for “New”
OSBS-Box are that a working OpenShift 4 cluster is readily available to deploy
into, you have a user account, **and you have access to the cluster admin
credentials** (unfortunately). That can be an OpenShift 4 cluster running in
someone’s cloud, on baremetal in your basement, or in a VM somewhere which is
hosting CRC (CodeReady Containers; basically the OS 4 equivalent of OS 3’s
`oc cluster`,  i.e. a single-node development cluster. I would *avoid* trying to run
CRC locally - its resource requirements are very high).

Everything is already configured to deploy to a default configured CRC cluster
instance. Cloning this repo as it stands and running the playbooks against a CRC
cluster should Just Work.

Note that these playbooks will run everything locally, so the `oc` command
**MUST** be available.

As mentioned, cluster admin is required, which means that you **must** override
'kubeadmin_pwd' in an 'overrides.yaml' file, in which you will also need to
override 'ocp_dev_passwd' with *your* OpenShift password, and provide your
dockerhub credentials as noted in 'docs/dockerhub.md'

OK, so there *are* some other assumptions here.

- Given that we might deploy into a running cluster somewhere, we want to be a
  good neighbor. “New” OSBS-Box will create Projects prefixed with (local)
  $USER, to avoid conflicts with existing Projects/Namespaces (you’ll need to put
  your OpenShift account name in ‘inventory.ini’ if it differs from $USER). This
  makes it possible for a team which is using a shared OpenShift 4 dev
  environment to run multiple deployments of OSBS in parallel.
- Backing storage is *not* created/configured. Simple Filesystem PVCs are
  requested, so if you don’t already have PVs which can provide that, you will
  need to set some up. Note that CRC provides 30 100G PVs by default.

Note that the README is hopelessly inaccurate and needs to be rewritten. I'm
holding off on that, on the assumption that discussion in this PR will result in
additional changes.

### CODE REVIEW

File: cleanup.yaml [hunk 1]
```diff
@@ -14,119 +15,63 @@
...
-      register: delete_koji_pvs
-      changed_when: delete_koji_pvs.rc == 0 and
-                    delete_koji_pvs.stdout != 'No resources found'
-      failed_when: delete_koji_pvs.rc != 0
+    - name: Delete koji project from openshift
+      command: oc delete project/"{{ koji_project }}"
+      register: delete_koji_project
+      changed_when: delete_koji_project.rc == 0
+      failed_when: delete_koji_project.rc != 0 and
+                   'cannot delete resource "projects"' not in delete_koji_project.stderr
```
Comment (mkosiarc) [inline cleanup.yaml]: Should this not be
`'cannot delete resource "projects"' in delete_koji_project.stderr` 
as when this message IS in the stderr, it defines failure? similiar in other instances

Comment (ben-alkov) [reply cleanup.yaml]: That's to cover the case where (as sometimes happens) the playbook is able to delete a Project, but is interrupted (or whatever) and fails to delete others. When the play is re-run, it will fail and exit because of that error on the already-deleted Project, thus skipping everything after that deletion, so we ignore it.

Comment (mkosiarc) [reply cleanup.yaml]: ack

**Review by mkosiarc — APPROVED** [review]

**Review by MartinBasti — APPROVED** [review]
Nice commits structure! :rocket:

**Review by lkolacek — APPROVED** [review]
Looks great!
